backend/src/Controller/ProductController.php

```diff
@@ -31,17 +31,40 @@ public function getAllProducts(ProductRepository $repository, Request $request):
             $products[] = $product;
         }
 
+        // $paginatedCollection
+        //     ->products = $products
+        //     ->total = $pagerfanta->getNbResults()
+        // ;
+        
+        // $route = 'products_collection';
+        // $routeParams = array();
+        // $createLinkUrl = function($targetPage) use ($route, $routeParams) {
+        //     return $this->generateUrl($route, array_merge(
+        //         $routeParams,
+        //         array('page' => $targetPage)
+        //     ));
+        // };
+
+        // $paginatedCollection->addLink('self', $createLinkUrl($page));
+        // $paginatedCollection->addLink('first', $createLinkUrl(1));
+        // $paginatedCollection->addLink('last', $createLinkUrl($pagerfanta->getNbPages()));
+
+        // if ($pagerfanta->hasNextPage()) {
+        //     $paginatedCollection->addLink('next', $createLinkUrl($pagerfanta->getNextPage()));
+        // }
+        // if ($pagerfanta->hasPreviousPage()) {
+        //     $paginatedCollection->addLink('prev', $createLinkUrl($pagerfanta->getPreviousPage()));
```

Comment:
is there a reason for leaving the comments?

Reply:
I was refactoring the code according to the review comments from Andreas, so this was work in progress adding a factory and services... Now its all cleaned up ;)
Anyway thank you for the comment!